devices/debian_wifi.py

```diff
@@ -1,88 +1,109 @@
         self.expect(self.prompt)
-        conn_state = self.before
-        match = re.search('Connected',conn_state)
+        match = re.search('Connected', self.before)
         if match:
-            return match.group(0)
+            return True
         else:
-            return None
+            return False
 
     def disconnect_wpa(self):
-        self.sendline("killall wpa_supplicant")
-        self.expect(self.prompt)
-
-    def wlan_ssid_connect(self, ssid_name):
-        self.sendline("iwconfig %s essid %s" % (self.iface_wlan,ssid_name))
-        self.expect(self.prompt)
-
-    def wlan_ssid_disconnect(self):
-        self.sendline("iw dev %s disconnect" % self.iface_wlan)
+        output = self.sudo_sendline("killall wpa_supplicant")
+        return output
+
+    def wlan_ssid_disconnect(self, iface):
+        output = self.sudo_sendline("iw dev %s disconnect" % iface)
+        return output
+
+    def wifi_disconnect(self, iface):
+        self.disconnect_wpa()
+        self.wlan_ssid_disconnect(iface)
+
+    def wifi_change_region(self, country):
+        country = countrycode(country, origin='country_name', target='iso2c')
+        self.sudo_sendline("iw reg set %s"%(country))
+        idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
+        if idx == 0:
+            self.sendline(self.password)
+            self.expect(self.prompt)
+        self.sendline("iw reg get")
         self.expect(self.prompt)
+        match = re.search(country, self.before)
+        if match:
+            return match.group(0)
+        else:
+            return None
 
-    def start_lan_client(self, wan_gw=None):
-        self.iface_dut = self.iface_wlan
+    def start_lan_client(self, iface):
+        self.iface_dut = iface
         super(DebianWifi, self).start_lan_client()
 
-    def linux_wifi_client(self, board, ssid_name, password=None):
+    def wifi_client_connect(self, iface, ssid_name, password=None):
         '''Scan for SSID and verify connectivity'''
-        output = self.scan()
-        match = re.search("%s" % ssid_name,output)
-        assert match!=None,'SSID value check in WLAN container'
-
-        link = self.link_up(self.iface_wlan)
-        if link==None:
-            wlan.set_link_state(self.iface_wlan)
-
-        if password:
-            conn_wpa = self.wpa_connect(ssid_name,password)
-            assert conn_wpa!=None,'WPA supplicant initiation'
-        else:
-            self.wlan_ssid_connect(ssid_name)
-
-        board.expect(pexpect.TIMEOUT, timeout=20)
-        conn_wlan = self.wlan_connectivity()
-        assert conn_wlan!=None,'Connection establishment in WIFI'
-
-        self.start_lan_client()
+        self.disable_and_enable_wifi(iface)
+        time.sleep(10)
+        output = self.wifi_check_ssid(iface,ssid_name)
+        assert output==True,'SSID value check in WLAN container'
+
+        conn_wifi = self.wifi_connect(iface,ssid_name,password)
+        time.sleep(20)
```

Comment:
nit: can we poll instead of just straight up sleeping? also self.expect(pexpect.TIMEOUT, timeout=20) is better than just sleeping

Reply:
Ack